sklearn/decomposition/tests/test_pca.py

```diff
@@ -389,6 +390,12 @@ def test_pca_validation():
                                     PCA(n_components, svd_solver=solver)
                                     .fit, data)
 
+    n_components = 1.2
```

Comment:
I might set n_components to something like 1.0 here because it seems a stronger test. But if you can make sure that your code work in such case, I think it's also acceptable.

Reply:
Ye, I didnt do it because before it was > 1 for it to apply. But will do it now since it's a closed set.